src/components/HeroFeatured/HeroFeatured.js

```diff
@@ -1,49 +1,93 @@
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+// import get all articles action
+import { getAllArticle } from '../../actions/article'
+
+// import component
+import HeroArticles from './HeroArticles';
+import MainFeature from './MainFeature';
+
 // styles
 import styles from './heroFeatured.module.scss';
 // images
 import thorAh from '../../authors_haven.png';
 
+
 class HeroFeatured extends Component {
   constructor(props) {
     super(props);
     this.state = {};
   }
 
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(getAllArticle());
+  }
+
   render() {
+    const data2 = [];
+    const { articles } = this.props;
+    const { data } = articles;
+    const newData = data.slice(0);
```

Comment:
I think you can skip line 31 and just have the array of articles directly in your articles prop by parsing it in your mapStateToProps method as:

```
const mapStateToProps = state => {
  const { allArticleReducer } = state;
  return {
    articles: allArticleReducer.data,
  }
}
```
This will give you access to the array of articles.